### ISSUE TITLE
warnings for ruby 2.7

### ISSUE THREAD

**Issue by SampsonCrowley** [brianmario/mysql2#1137]
```
gems/mysql2-0.5.3/lib/mysql2/client.rb:131: warning: rb_tainted_str_new_cstr is deprecated and will be removed in Ruby 3.2
```

I ran into this when running the rails code test suite, and I don't use mysql; so I don't know exactly what's causing it, but running `_query` in `client.rb` uses a deprecated `C` method for string manipulation.

A similar warning was removed from the `pg` gem in `1.2.0`

**Comment by SampsonCrowley** [comment]
@here no comment?

**Comment by sodabrew** [comment]
Thanks for the heads up. It's a warning and you will not be impacted at this time.

**Comment by sodabrew** [comment]
Resolved by 785969fbceadd96d495ec9fecc67efee03d0056c

**Comment by SampsonCrowley** [comment]
@sodabrew if that PR resolves this issue was merged almost a year ago why are we still being overloaded with warnings in every library that uses mysql2 for ruby 2.7? It's seriously to the point where terminal logs are unreadable from so many warning calls without purposefully silencing this gem

**Comment by SampsonCrowley** [comment]
![Screenshot from 2020-09-22 13-19-36](https://user-images.githubusercontent.com/22461638/93928008-e5fd3900-fcd6-11ea-8b80-f5436a389523.png)
![Screenshot from 2020-09-22 13-22-32](https://user-images.githubusercontent.com/22461638/93928020-e990c000-fcd6-11ea-838e-66e8a458e17f.png)
![Screenshot from 2020-09-22 13-23-24](https://user-images.githubusercontent.com/22461638/93928025-ec8bb080-fcd6-11ea-95d9-1067f0235345.png)

Certainly feels like I'm being impacted, not being able to easily scroll back through test output without wading through hundreds of warnings

**Comment by sodabrew** [comment]
Ah, I wasn't aware this was a runtime warning. I thought it was compile time only.

I'll prioritize putting out a new release that includes the fix from last year.

**Comment by SampsonCrowley** [comment]
@sodabrew thank you very much, I appreciate it

**Comment by tisba** [comment]
Gentle ping @sodabrew :)

**Comment by jasonperrone** [comment]
Doesn't seem very prioritized.

**Comment by lucy-dement** [comment]
Hi,
Is there any update about when this will be released? End of life for ruby 2.6 is in a couple of months and it would be nice to fix all the deprecation warnings in my application while doing the upgrade.
Thanks
